sfge/src/test/java/com/salesforce/graph/visitor/PathScopeVisitorTest.java

```diff
@@ -110,4 +112,22 @@ public void testParameterNameShadowsField() {
         MatcherAssert.assertThat(
                 TestUtil.apexValueToString(value.getValues().get(1)), equalTo("value2"));
     }
+
+    @Test
+    public void testVariableNameReuseThrowsUserActionException() {
```

Comment:
Could we add a test case here to confirm that the `this.whatever = whatever` case throws no exception? (Unless we already have coverage for that. Do we?)

Reply:
Good idea. Will add the test while we remember.